charts/karpenter/templates/controller/serviceaccount.yaml

```diff
@@ -0,0 +1,12 @@
+{{- if .Values.controller.serviceAccount.create -}}
```

Comment:
since we now have a controller service account and a webhook service account, the IAM Roles for Service Accounts docs would need to be updated to create both or instructions to annotate both. I'm not sure breaking these apart is worth the extra setup work. 

I actually think we could just combine the controller and webhook containers into the same pod. There would be multiple services for the pod but that's doable with the named ports. This would simplify a lot of the redundant parameters between the webhook and controller in the values.yaml. WDYT?

Reply:
The K8s roles were already separate, so it would be correct to split the IAM roles; but duplicating the IRSA annotation would make it behave as it currently does. Merging the pods would probably be best in the long run.

Reply:
Did you want to take on the merging of the webhook and controller to one pod in order to simplify the helm chart?

Reply:
Could we leave that until this has been merged as it's adding another level of complexity or do you want all of the pain now?

Reply:
I think it would simplify the helm chart quite a bit (getting rid of the controller and webhook block, removing service account duplication, etc), so I'd advocate for all the pain now :D

Reply:
OK, I'll do it.

Reply:
@bwagner5 I've just done this, and the chart does look a lot cleaner.

Reply:
nice! so much cleaner! Thanks for doing that!!